### DISCUSSION TITLE
Jupyter on "externally-managed-environment"

### DISCUSSION THREAD

**Discussion by agerlach** [quarto-dev/quarto-cli#6572] [Q&A] [answered]
### Description

Is there a recommended way to install Jupyter and Jupyter Cache on an OS that implements PEP 668 (externally-managed-environment) for usage with Quarto?

I am using a Debian 12 docker container for building quarto documents, however the OS now prevents you from doing a global `pip install`. It seems that the preferred method to install is using `pipx`

```
pipx ensurepath 
pipx install jupyter --include-deps 
pipx inject jupyter jupyter-cache
```

Running `which jupyter` returns `/root/.local/bin/jupyter`. However `quarto check jupyter` returns

```
[✓] Checking Python 3 installation....OK
      Version: 3.11.2
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter
```

So, despite being on the PATH it is not being picked up. I realize that I could use venv, but I don't actually use Python for my documents, so I don't want to have to carry a .venv for each project. Thanks.

- Quarto version: 1.4.322
- OS: Debian 12 (Bookworm)

**Comment by mcanouil** [comment]
If you are not using Jupyter, are you using knitr?
You can enforce an engine with `engine: knitr`, see <https://quarto.org/docs/computations/execution-options.html#engine-binding>.

**Reply by agerlach** [reply]
I am using Jupyter but for Julia.

**Comment by agerlach** [comment, marked as answer]
Being that this is a docker container, I guess I don't really need to worry about messing with the global python setup. So, I am just going to set `ENV PIP_BREAK_SYSTEM_PACKAGES=1 ` ([details](https://veronneau.org/python-311-pip-and-breaking-system-packages.html))
